tests/CMakeLists.txt

```diff
@@ -1,5 +1,4 @@
-project(ensmallen_tests CXX)
```

Comment:
Out of curiosity, why did you unset the project variable for this?

Reply:
I looked at a couple other projects to get a feel of how it's written, like [type-safe](https://github.com/foonathan/type_safe/) and [cxxopts](https://github.com/jarro2783/cxxopts/) and they did it the same way. It seemed to me that a project was defined as the full library, and the testing subdirectory didn't seem to qualify by itself. I'm not sure whether there is any gain to removing the project variable though, I'll add it back if that's what you suggest.

Reply:
Following such a convention seems reasonable to me. As long as we follow some standard, right? :) I just worry if the repitition of the literal string could lead to more errors down the line.

Reply:
Yeah, personally I agree with @favre49 here; I think just having it as `add_executable` is just fine.